src/axom/primal/geometry/KnotVector.hpp

```diff
@@ -128,25 +258,88 @@ class KnotVector
     }
   }
 
-  /// \brief Insert a knot into the knot vector r times
-  void insertKnot(axom::IndexType span, T t, int r = 1)
+  /// \brief Rescale the knot vector to the span of [a, b]
+  void rescale(T a, T b)
+  {
+    T min_knot = m_knots[0];
+    T max_knot = m_knots[m_knots.size() - 1];
+    T span = max_knot - min_knot;
+
+    for(int i = 0; i < m_knots.size(); ++i)
+    {
+      m_knots[i] = a + (m_knots[i] - min_knot) * (b - a) / span;
+    }
+  }
```

Comment:
Gut check -- does this work when $a == b$ or when $a > b$?
Please add tests and/or assertions.

Also -- what happens if $min_knot >= max_knot$

Reply:
It shouldn't work, the knot vector wouldn't be monotonic in either case. I'm going to check this with asserts.